app/controllers/submit.js

```diff
@@ -958,6 +934,128 @@ export default Controller.extend(Analytics, BasicsValidations, NodeActionsMixin,
 
             run.later(() => context.$(`#${elementId}`).removeClass(highlightClass), 2000);
         },
+        currentUserRemoved() {
+            this.send('cancel');
+        },
+
+        /*
+        Supplemental Project Section
+        */
+        changeSupplementalPickerState(state) {
+            // Sets supplementalPickerState to edit, start, new, or existing
+            this.set('supplementalPickerState', state);
+            this.send('discardSupplemental');
+            if (this.get('firstSupplementalOpen') && state === this.get('_State').NEW) {
+                // Populates the supplemental project title the first time
+                // "Create a new OSF Project" is selected
+                this.set('pendingSupplementalProjectTitle', `Supplemental materials for ${this.get('currentProvider.documentType.singular')}: ${this.get('model.title')}`);
+                this.set('firstSupplementalOpen', false);
+                this.set('supplementalProjectTitleValid', true);
+            }
+        },
+
+        setSupplementalTitleFromSelected() {
+            // Locally sets supplemental project title in UI - title taken from existing project.
+            // When preprint submission is finalized, the selectedSupplemental project
+            // will be set as the supplemental project on the preprint.
+            this.get('metrics')
+                .trackEvent({
+                    category: 'button',
+                    action: 'click',
+                    label: 'Submit - Set existing project as supplemental',
+                });
+            this.set('supplementalProjectTitle', this.get('selectedSupplementalProject.title'));
+            this.set('pendingSupplementalProjectTitle', '');
+            this.set('node', this.get('selectedSupplementalProject'));
+            this._moveFromSupplemental();
+        },
+
+        setSupplementalProjectTitle() {
+            // Locally sets supplementalProjectTitle - Submit mode
+            // When preprint submission is finalized, a supplemental project will
+            // be created with this title.
+            this.get('metrics')
+                .trackEvent({
+                    category: 'button',
+                    action: 'click',
+                    label: 'Submit - Set new project as supplemental',
+                });
+            this.set('supplementalProjectTitle', this.get('pendingSupplementalProjectTitle'));
+            this.set('selectedSupplementalProject', null);
+            this.set('node', null);
+            this._moveFromSupplemental();
+        },
+
+        discardSupplemental() {
+            // Locally resets supplemental project UI to latest save state
```

Comment:
Lots of logic here, but this is also needed because the Submit mode and Edit mode supplemental states behave pretty differently.